libbeat/common/kubernetes/metadata/node.go

```diff
@@ -0,0 +1,70 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package metadata
+
+import (
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/kubernetes"
+)
+
+type node struct {
+	store    cache.Store
+	resource *Resource
+}
+
+// NewNodeMetadataGenerator creates a metagen for service resources
+func NewNodeMetadataGenerator(cfg *common.Config, nodes cache.Store) MetaGen {
+	no := &node{
```

Comment:
nit. Could we avoid using this extra variable?
```suggestion
	return node{...}
```
see https://github.com/elastic/beats/blob/d7d2c177120e70676df490ddf3fbfc899f8dce67/libbeat/common/kubernetes/metadata.go#L58

Reply:
i initially designed the New with an error type being returned. i had later changed my mind but didnt bother removing the temp var.

Reply:
👍